xrmocap/ops/bottom_up_association/associate/fourdag_associate.py

```diff
@@ -0,0 +1,913 @@
+        self.c_view_cnt = c_view_cnt
+        self.min_check_cnt = min_check_cnt
+        self.min_asgn_cnt = min_asgn_cnt
+        self.normalize_edges = normalize_edges
+
+        self.paf_dict = skel_info[kps_convention]['paf_dict']
+        self.hierarchy_map = skel_info[kps_convention]['hierarchy_map']
+
+        self.m_paf_hier = np.zeros(self.n_pafs)
+        for paf_id in range(self.n_pafs):
+            self.m_paf_hier[paf_id] = min(
+                self.hierarchy_map[self.paf_dict[0][paf_id]],
+                self.hierarchy_map[self.paf_dict[1][paf_id]])
+        self.m_paf_hier_size = self.m_paf_hier.max()
+
+        self.m_joint2paf = {i: [] for i in range(self.n_kps)}
+        for paf_id in range(self.n_pafs):
+            joint_pair = [self.paf_dict[0][paf_id], self.paf_dict[1][paf_id]]
+            self.m_joint2paf[joint_pair[0]].append(paf_id)
+            self.m_joint2paf[joint_pair[1]].append(paf_id)
+
+        self.m_epi_edges = {
+            i: {
+                j: {k: -1
+                    for k in range(self.n_views)}
+                for j in range(self.n_views)
+            }
+            for i in range(self.n_kps)
+        }
+        self.m_tempEdges = {
+            i: {j: -1
+                for j in range(self.n_views)}
+            for i in range(self.n_kps)
+        }
+        self.m_joint_rays = {
+            i: {j: []
+                for j in range(self.n_kps)}
+            for i in range(self.n_views)
+        }
+        self.m_bone_nodes = {
+            i: {j: []
+                for j in range(self.n_views)}
+            for i in range(self.n_pafs)
+        }
+        self.m_bone_epi_edges = {
+            i: {
+                j: {k: []
+                    for k in range(self.n_views)}
+                for j in range(self.n_views)
+            }
+            for i in range(self.n_pafs)
+        }
+        self.m_bone_temp_edges = {
+            i: {j: []
+                for j in range(self.n_views)}
+            for i in range(self.n_pafs)
+        }
+        self.m_assign_map = {
+            i: {j: []
+                for j in range(self.n_kps)}
+            for i in range(self.n_views)
+        }
+        self.m_persons_map = dict()
+
+        self.cameras = []
+        self.last_multi_kps3d = dict()
+        self.cliques = []
+
+    def set_cameras(self, cameras_param):
+        for view in range(len(cameras_param)):
+            self.cameras.append(Camera(cameras_param[view]))
+        self.n_views = len(cameras_param)
+
+    def __call__(self, kps2d, pafs, last_multi_kps3d=dict):
+        """Match people id from different cameras.
+
+        Args:
+
+        Raises:
+
+        Returns:
+        """
+        self.kps2d = kps2d
+        self.pafs = pafs
+        self.last_multi_kps3d = last_multi_kps3d
+        self.construct_graph()
+        self.solve_graph()
+        
+        return self.m_persons_map
+
+    def construct_graph(self):
+        self.__calculate_joint_rays()
```

Comment:
After checking PEP8, I find that we abuse `_` a lot in XRMoCap. Below is the guide from PEP8, you can also find it at [python.org](https://peps.python.org/pep-0008/#descriptive-naming-styles):
```
In addition, the following special forms using leading or trailing underscores are recognized (these can generally be combined with any case convention):

_single_leading_underscore: weak “internal use” indicator. E.g. from M import * does not import objects whose names start with an underscore.
```